ql/experimental/callablebonds/callablebond.cpp

```diff
@@ -108,6 +109,270 @@ namespace QuantLib {
     }
 
 
+    /* Expose a mechanism to add spread the engine parameters and
+       clean clean up when the object is destroyed
+     */
+    class EngSpreadHelper {
+        CallableBond::arguments*  args;
+        double ospread;
+    public:
+        explicit EngSpreadHelper(CallableBond::arguments* args):
+            args(args),
+            ospread(args->spread)
+        {
+        }
+
+        void setSpread(double s)
+        {
+            args->spread=s;
+        }
+
+        ~EngSpreadHelper()
```

Comment:
This is required because `CallableBond::setupArguments` doesn't reset the spread. It could be deleted if it did. However, see below.